Test Suite Implementation

matjames007 opened the pull request:
The intention is to create a test suite that is capable of covering the testing of the API, the front-end and most of the functions written for the platform.  The test suite will be run for each build by travis using `mocha` testing framework with support from `supertest` for end-point testing and `protractor` for angular based tests.

matjames007 committed:
I am unsure about the effects this will have on travis since the environment variables are not set.
I expect that this will not work nicely.

Regards,

-Matthew Budram

matjames007 committed:
The changes here now allow mocha to trigger a loading of the express application that was
missing in the previous deployment during testing.

Regards,

-Matthew Budram

matjames007 committed:
I found a bug in the removal of the test data which fails due to an invalid reference.  This should fix things now.

Regards
-Matthew Budram

paradoxxjm commented on package.json:
```diff
@@ -4,12 +4,14 @@
+    "chai": "^3.4.0",
     "connect-flash": "^0.1.1",
     "cookie-parser": "~1.3.4",
     "coveralls": "^2.11.2",
```
Coveralls reference in package.json

matjames007 commented:
Things to do with this PR:
-  Add protractor test cases
-  Give additional RESTful based examples
-  Develop Authentication test scripts
-  Add unit tests (especially for complicated functions)
-  Coveralls Integration

matjames007 committed:
A restructure was done to the initial test-suite.  This is done in preparation for Testing Thursdays!

Matthew Budram

matjames007 committed:
Made a few modifications to demonstrate data integrity checks on the data being persisted.

Matthew Budram

matjames007 committed:
I added a few checks for the farmer endpoint of the API.  I also made modifications
to the existing UNIT entity as there existed the potential for the removal to be
completed before the actual testing.

Matthew Budram

matjames007 committed:
I found a bug with the manipulation of the response object within the test suite.

This will always cause one of the test cases to fail.  This commit should resolve
it now.

Regards,
-Matthew Budram
